clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -506,18 +504,34 @@ static CallInst *CreateBuiltinCallWithAttr(CodeGenFunction &CGF,
 
 // Emit a simple mangled intrinsic that has 1 argument and a return type
 // matching the argument type. Depending on mode, this may be a constrained
-// floating-point intrinsic.
+// or an fpbuiltin floating-point intrinsic.
 static Value *emitUnaryMaybeConstrainedFPBuiltin(CodeGenFunction &CGF,
                                 const CallExpr *E, unsigned IntrinsicID,
                                 unsigned ConstrainedIntrinsicID) {
   llvm::Value *Src0 = CGF.EmitScalarExpr(E->getArg(0));
-  if (CGF.CGM.getCodeGenOpts().FPAccuracy) {    
-    Function *Func =
-        CGF.CGM.getIntrinsic(ConstrainedIntrinsicID, Src0->getType());
-    return CreateBuiltinCallWithAttr(CGF, Func, {Src0},
+  if (CGF.CGM.getCodeGenOpts().FPAccuracy) {
```

Comment:
We need some way to handle the case where fp-accuracy is combined with a mode that requires constrained intrinsics. We should probably reject any such combinations in the driver, but it can also happen with pragmas, so we'll need to check it in the pragma handling too.

Eventually I'd like to have the fpbuiltin intrinsics take rounding mode and exception behavior attributes, but that is beyond the current scope.

Reply:
> We need some way to handle the case where fp-accuracy is combined with a mode that requires constrained intrinsics. We should probably reject any such combinations in the driver, but it can also happen with pragmas, so we'll need to check it in the pragma handling too.
> 
Can you give an example please?
> Eventually I'd like to have the fpbuiltin intrinsics take rounding mode and exception behavior attributes, but that is beyond the current scope.
That's fine. We could do that. These attributes can be added in to the FPAccuracyMap, in which case it will have more than one entry.